### PR TITLE
[1-0002: Create infrastructure for project] Controller Initialization

### PR DESCRIPTION
**Pull request by dencheek1**
A simple test controller was created.
Spring version variable added in main pom.xml.
New test controller and it's configuration.

### CODE REVIEW

File: epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <load-on-startup>1</load-on-startup>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>dispatcher</servlet-name>
+        <url-pattern>/*</url-pattern>
+    </servlet-mapping>
+
+
+</web-app>
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml]: add empty line at the end of the file

File: epam-vote-system-controller/src/main/webapp/WEB-INF/dispatcher-servlet.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+                           http://www.springframework.org/schema/beans/spring-beans.xsd
+                           http://www.springframework.org/schema/context
+                           http://www.springframework.org/schema/context/spring-context.xsd">
+
+    <context:component-scan base-package="com.epam.vote.controller"/>
+
+</beans>
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/dispatcher-servlet.xml]: the same - empty line

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.epam.vote.controller;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Dzianis Zatsiupa
+ */
+
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: remove empty line

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: the same remove empty line

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+@RestController
+@RequestMapping("/api")
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: general URL shouldn't be placed in each controllers. Move it into web.xml

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.epam.vote.controller;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Dzianis Zatsiupa
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java:7]: Let's write comment in the next formatting:
/**
 * "Description of class".
 * <p/>
 * Copyright (C) 2021
 * <p/>
 * Date: "CURRENT DATE"
 * @author "YOUR NAME"
 */

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/ContextConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @author Dzianis Zatsiupa
+ *
+ * */
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {"com.epam.vote.controller"})
+public class ContextConfiguration {
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/ContextConfiguration.java]: remove this class and use xml file

File: epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
+                      http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
+         version="4.0">
+
+    <display-name>api</display-name>
+
+    <context-param>
+        <param-name>contextClass</param-name>
+        <param-value>org.springframework.web.context.support.AnnotationConfigWebApplicationContext</param-value>
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml]: redundant block

File: epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
+                      http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
+         version="4.0">
+
+    <display-name>api</display-name>
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml]: Vote

File: pom.xml [hunk 10]
```diff
@@ -23,6 +23,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <checkstyle-version>3.0.0</checkstyle-version>
+        <spring.versoin>5.3.18</spring.versoin>
```
Comment (AntonAzarenka) [inline pom.xml]: check spelling

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.epam.vote.controller;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * "Show hello world message"
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: Shows

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (C) 2021
+ *
+ * Date:"04.06.2022"
+ * @author Dzianis Zatsiupa
+ * */
+
+@RestController
+public class TestController {
+
+    @RequestMapping("*")
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: /

File: epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/web.xml:1]: move this file to resource folder

File: epam-vote-system-controller/src/main/webapp/WEB-INF/dispatcher-servlet.xml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AzarenkaA) [inline epam-vote-system-controller/src/main/webapp/WEB-INF/dispatcher-servlet.xml]: rename this file like "spring-mvc.xml

File: epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (C) 2017 epam.com
+ *<p/>
+ * Date:04/07/2022
+ *
+ * @author Dzianis Zatsiupa
+ */
+@RestController
+public class TestController {
+
+    @RequestMapping("/*")
```
Comment (AntonAzarenka) [inline epam-vote-system-controller/src/main/java/com/epam/vote/controller/TestController.java]: why do we need * in the path?

**Pull request merged**